### ISSUE TITLE
Data type error 

### ISSUE THREAD

**Issue by zer0trust** [microsoft/Requirements#9]
```
[PS C:\users\zer0trust\Desktop> $requirements | Invoke-Requirement | Format-Checklist
Format-Checklist : The input object cannot be bound to any parameters for the command either because the command does not take pipeline input or the input and its properties do not match any of the parameters that take pipeline input.
At line:1 char:38
+ $requirements | Invoke-Requirement | Format-Checklist
+                                      ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (RequirementEvent:RequirementEvent) [Format-Checklist], ParameterBindingException
    + FullyQualifiedErrorId : InputObjectNotBound,Format-Checklist]
```

```
PS C:\users\zer0trust\Desktop> $results = $requirements | Invoke-Requirement
PS C:\users\zer0trust\Desktop> $results.gettype()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     Object[]                                 System.Array
```

The Format-Checklist function appears to be looking for a "RequirementEvent" data type from the output of Invoke-Requirement, but the output of the commands show that it has the System.Array type instead.

**Comment by chriskuech** [comment]
I'm having trouble reproing this one.  Can you try running `example.ps1` in the module root and seeing if you get any errors?

**Comment by zer0trust** [comment]
Example.ps1 does run without any errors.

**Comment by chriskuech** [comment]
Can you repro anything in a clean session?

**Comment by zer0trust** [comment]
Clean session on a different machine: 

```
PS C:\WINDOWS\system32> get-module requirements
PS C:\WINDOWS\system32> import-module requirements
PS C:\WINDOWS\system32> get-module requirements

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     2.2.1      requirements                        {Format-CallStack, Format-Checklist, Invoke-Requirement, N...


PS C:\WINDOWS\system32> $requirements = @{
>>
>>   Name = "File-Add"
>>
>>   Describe = "Add test.txt file"
>>
>>   Test = {
>>     test-path -Path "c:\test.txt"
>>   }
>>   Set = {
>>      new-item -Path "c:\test.txt"
>>   }
>> }
PS C:\WINDOWS\system32> $requirements

Name                           Value
----                           -----
Name                           File-Add
Set                               ...
Describe                       Add test.txt file
Test                              ...


PS C:\WINDOWS\system32> $requirements.gettype()

IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     Hashtable                                System.Object


PS C:\WINDOWS\system32> $requirements | Invoke-Requirement


Date        : 7/2/2019 9:18:57 AM
Method      : Test
State       : Start
Result      :
Requirement : File-Add

Date        : 7/2/2019 9:18:57 AM
Method      : Test
State       : Stop
Result      : False
Requirement : File-Add

Date        : 7/2/2019 9:18:58 AM
Method      : Set
State       : Start
Result      :
Requirement : File-Add

LastWriteTime : 7/2/2019 9:18:58 AM
Length        : 0
Name          : test.txt

Date        : 7/2/2019 9:18:58 AM
Method      : Set
State       : Stop
Result      : False
Requirement : File-Add

Date        : 7/2/2019 9:18:58 AM
Method      : Validate
State       : Start
Result      :
Requirement : File-Add

Date        : 7/2/2019 9:18:58 AM
Method      : Validate
State       : Stop
Result      : True
Requirement : File-Add



PS C:\WINDOWS\system32> $requirements | Invoke-Requirement | Format-Checklist
09:19:06 [   ] Add test.txt file
PS C:\WINDOWS\system32> $requirements | Invoke-Requirement | Format-Checklist
09:19:29 [   ] Add test.txt fileFormat-Checklist : The input object cannot be bound to any parameters for the command either because the command does
not take pipeline input or the input and its properties do not match any of the parameters that take pipeline input.
At line:1 char:38
+ $requirements | Invoke-Requirement | Format-Checklist
+                                      ~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (C:\test.txt:PSObject) [Format-Checklist], ParameterBindingException
    + FullyQualifiedErrorId : InputObjectNotBound,Format-Checklist
```

**Comment by chriskuech** [comment]
The `Set` `scriptblock` is supposed to log the result to the `RequirementEvent` just like `Test` `scriptblock`s.  Because the expression was not captured, it was being outputted and implicitly casted to a `RequirementEvent`, causing this type error.  This should be fixed in #14 .
